packages/camera/lib/src/common/native_texture.dart

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+
+import 'camera_channel.dart';
+import 'camera_mixins.dart';
+
+/// Used to allocate a buffer for displaying a preview camera texture.
+///
+/// The [textureId] can be passed to a [Texture] widget.
+class NativeTexture with CameraClosable, CameraMappable {
```

Comment:
This is used to for a developer to have a control over `TextureRegistry.SurfaceTextureEntry`. It's so they can have direct access to the textureId and reuse it for different camera instances.